src/giskardpy/visualization.py

```diff
@@ -0,0 +1,67 @@
+from plugin import GiskardBehavior
+import py_trees
+#import roslib; roslib.load_manifest('urdfdom_py')
+from urdf_parser_py.urdf import URDF
+from interactive_markers.menu_handler import *
+from visualization_msgs.msg import *
+from symengine_robot import hacky_urdf_parser_fix
+
+
+class VisualizationBehavior(GiskardBehavior):
+    def __init__(self, name):
+        super(VisualizationBehavior, self).__init__(name)
+
+    def setup(self, timeout):
+        self.robot_d = rospy.get_param(u'/robot_description')
+        self.robot = URDF.from_xml_string(hacky_urdf_parser_fix(self.robot_d))
+        self.publisher = rospy.Publisher('giskard_visualization', MarkerArray, queue_size=1)
+        self.num_markers = len(self.robot.links)
+        self.robot_base = self.robot.get_root()
+        self.enable_visualization = False
+        if rospy.has_param(u'/giskard/enable_visualization'):
```

Comment:
Read the ros params in giskard_trees and pass them as paramters in the __init__. That way one can easily find where all the rosparams are parsed.